commands/preuninstall.ts

```diff
@@ -0,0 +1,33 @@
+﻿///<reference path="../../.d.ts"/>
+"use strict";
+
+import options = require("../options");
+import path = require("path");
+import util = require("util");
+
+export class PreUninstallCommand implements ICommand {
+	private static ADB_RELATIVE_PATH = "../resources/platform-tools/android/%s/adb";
+
+	constructor(private $fs: IFileSystem,
+		private $childProcess: IChildProcess) { }
+	public disableAnalytics = true;
+
+	public allowedParameters: ICommandParameter[] = [];
+
+	public execute(args: string[]): IFuture<void> {
+		return (() => {
+			var relativeAdbPath = util.format(PreUninstallCommand.ADB_RELATIVE_PATH, process.platform);
+			var adbPath = path.join(__dirname, relativeAdbPath);
+			var executableAdbPath = process.platform === "win32" ? adbPath + ".exe" : adbPath;
+			if(!this.$fs.exists(executableAdbPath).wait()) {
+				// in some older installations adb was not in common
+				adbPath = path.join(__dirname, "..", "..", relativeAdbPath);
+			}
+
+			var killAdbServerCommand = util.format("\"%s\" kill-server", adbPath);
```

Comment:
I think different clean-up operation should be guarded in try/catch, so if one of them fails, others have chance to execute